app/shared/app-platform/src/androidMain/kotlin/platform/MeteredNetworkDetector.android.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2024 OpenAni and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license, which can be found at the following link.
+ *
+ * https://github.com/open-ani/ani/blob/main/LICENSE
+ */
+
+package me.him188.ani.app.platform
+
+import android.content.BroadcastReceiver
+import android.content.Intent
+import android.content.IntentFilter
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+
+
+private class AndroidMeteredNetworkDetector(
+    private val context: Context
+) : BroadcastReceiver(), MeteredNetworkDetector {
+    private val flow = MutableSharedFlow<Boolean>(extraBufferCapacity = 1)
```

Comment:
```suggestion
    private val flow = MutableStateFlow<Boolean>(false)
```
这必须保证会 emit 一个值, 否则 `combine(anitorrentConfigFlow, isMeteredNetworkFlow) { config, isMetered ->` 会永远不 emit